srcs/app/server/Connection.cpp

```diff
@@ -1,32 +1,40 @@
 #include <Connection.hpp>
-#include <iostream>
+#include <HttpRequestHandler.hpp>
 
-Connection::Connection(int socket) : socket_(socket) {
-	// Temporary, until we implement HttpRequest and HttpResponse
-	raw_response_ = "HTTP/1.1 200 OK\r\n"
-								  "Server: hello_world\r\n"
-								  "Content-Length: 22\r\n"
-								  "Content-Type: text/html\r\n"
-								  "\r\n"
-								  "<h1>Hello, World!</h1>";
+Connection::Connection(const ServerConfig &server_config, int socket)
+	: server_config_(server_config), socket_(socket),
+		ready_for_response_(false), keep_alive_(true),
+		raw_request_(""), raw_response_("") {
 }
 
 bool	Connection::ReadRequest() {
-	// Temporary solution, until we implement the HttpRequest class
 	char	buffer[4096];
 
 	int nbytes = recv(socket_, buffer, sizeof(buffer), 0);
 	if (nbytes <= 0)
 		return false;
+	ready_for_response_ = true;
 	raw_request_.append(&buffer[0], &buffer[nbytes]);
 	return true;
 }
 
 bool	Connection::SendResponse() {
-	// Temporary solution, until we implement the HttpResponse class
+	if (!ready_for_response_)
+		return true;
+	if (raw_response_.empty()) {
+		HttpRequestHandler	handler(server_config_, raw_request_);
```

Comment:
Maybe this still could be a little coupled? What about making an interface to enable interaction between server and implemented protocol, in this case, http? This is just an idea. You could have a generic protocol interface which defines interaction methods and, then, instantiate the httpHandler through an interface pointer. This could be a good point to start using interfaces. As I said, just an idea

Reply:
I like this idea :100:. I've added IRequestHandler interface.